modules/rostests/apitests/ntdll/LdrLoadDll.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Test for LdrLoadDll
+ * COPYRIGHT:   Copyright 2017 Thomas Faber <thomas.faber@reactos.org>
+ *              Copyright 2024 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include "precomp.h"
+
+/*
+
+NTSTATUS
+NTAPI
+LdrLoadDll(
+    IN PWSTR SearchPath OPTIONAL,
+    IN PULONG DllCharacteristics OPTIONAL,
+    IN PUNICODE_STRING DllName,
+    OUT PVOID *BaseAddress
+);
+
+*/
+
+START_TEST(LdrLoadDll)
+{
+    NTSTATUS Status = STATUS_SUCCESS;
+    UNICODE_STRING DllName;
+    PVOID BaseAddress, BaseAddress2;
+    WCHAR szWinDir[MAX_PATH], szSysDir[MAX_PATH], szPath[MAX_PATH];
+    WCHAR *pch;
+
+    RtlInitEmptyUnicodeString(&DllName, NULL, 0);
+    GetWindowsDirectoryW(szWinDir, _countof(szWinDir));
+    GetSystemDirectoryW(szSysDir, _countof(szSysDir));
+
+    Status = 0xDEADFACE;
+    StartSeh()
+        Status = LdrLoadDll(NULL, NULL, &DllName, NULL);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+    ok_long(Status, 0xDEADFACE);
+
+    Status = 0xDEADFACE;
+    BaseAddress = InvalidPointer;
+    StartSeh()
+        Status = LdrLoadDll(NULL, NULL, &DllName, &BaseAddress);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+    ok_long(Status, 0xDEADFACE);
+    ok_ptr(BaseAddress, InvalidPointer);
+
+    Status = 0xDEADFACE;
+    StartSeh()
+        Status = LdrLoadDll(L"", NULL, &DllName, NULL);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+    ok_long(Status, 0xDEADFACE);
+
+    Status = 0xDEADFACE;
+    BaseAddress = InvalidPointer;
+    StartSeh()
+        Status = LdrLoadDll(L"", NULL, &DllName, &BaseAddress);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+    ok_long(Status, 0xDEADFACE);
+    ok_ptr(BaseAddress, InvalidPointer);
+
+    RtlInitUnicodeString(&DllName, L"advapi32.dll");
+
+    StartSeh()
+        Status = LdrLoadDll(NULL, NULL, &DllName, NULL);
+        if (NT_SUCCESS(Status))
+            LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_ACCESS_VIOLATION);
```

Comment:
No ok_long and ok_ptr tests as above? (even if there's the status_access_violation raised?)

Reply:
tests added.